pkg/webhook/webhook.go

```diff
@@ -99,6 +116,37 @@ func (s *Server) handleWebhook() http.Handler {
 
 		if err = s.eventsPubsub.Send(context.Background(), eventBytes); err != nil {
 			logger.Errorw("failed to write messages to event pubsub", "code", http.StatusInternalServerError, "body", errWritingToBackend, "error", err)
+
+			exceeds, bqQueryErr := s.bqClient.
+				FailureEventsExceedsRetryLimit(ctx, s.failureEventTableID, deliveryID, s.retryLimit)
+			if bqQueryErr != nil {
+				logger.Errorw("failed to call BigQuery", "method", "FailureEventsExceedsRetryLimit",
+					"code", http.StatusInternalServerError, "body", errWritingToBackend, "error", bqQueryErr)
+			} else if exceeds {
+				// exceeds the limit, write to DLQ
+				if errDLQ := s.dlqEventsPubsub.Send(context.Background(), eventBytes); errDLQ != nil {
```

Comment:
Not sure why you need a different name for error
```suggestion
				if err := s.dlqEventsPubsub.Send(context.Background(), eventBytes); err != nil {
```